core/consensus/grandpa/structs.hpp

```diff
@@ -7,27 +7,28 @@
 #define KAGOME_CORE_CONSENSUS_GRANDPA_STRUCTS_HPP
 
 #include <boost/asio/steady_timer.hpp>
-#include <boost/operators.hpp>
+#include <boost/variant.hpp>
 #include "common/blob.hpp"
+#include "common/visitor.hpp"
 #include "common/wrapper.hpp"
+#include "consensus/grandpa/common.hpp"
 #include "crypto/ed25519_types.hpp"
 #include "primitives/authority.hpp"
 #include "primitives/common.hpp"
 
 namespace kagome::consensus::grandpa {
 
-  using Timer = boost::asio::basic_waitable_timer<std::chrono::steady_clock>;
+  using Timer = boost::asio::basic_waitable_timer<std::chrono::system_clock>;
 
-  using RoundNumber = common::Wrapper<uint64_t, struct RoundNumberTag>;
+  using BlockInfo = primitives::BlockInfo;
+  using Precommit = primitives::detail::BlockInfoT<struct PrecommitTag>;
+  using Prevote = primitives::detail::BlockInfoT<struct PrevoteTag>;
+  using PrimaryPropose =
+      primitives::detail::BlockInfoT<struct PrimaryProposeTag>;
 
-  /// voter identifier
-  using Id = primitives::AuthorityId;
-
-  using BlockHash = primitives::BlockHash;
-  using BlockNumber = primitives::BlockNumber;
-
-  /// voter signature
-  using Signature = crypto::ED25519Signature;
+  const static uint8_t kPrevoteStage = 0;
```

Comment:
They are defined in spec. These are the type ids for the different vote types. They used for serialization and signing.

It might be a enum, but then I would need to define << and >> operators for it. I was lazy to do that and left as is